### ISSUE TITLE
Masonry + imagesLoaded + isFitWidth: true shows items vertically on one single line

### ISSUE THREAD

**Issue by andru26** [desandro/masonry#953]
Hello, 

I am trying to implement your solution for loading the images one by one and it doesn't work well with layoutMode: 'masonry', and isFitWidth: true

I setup an example based on your code and I would like to know how can I make the items filll in the entire container with my current setup. Here is my code:

http://codepen.io/andru26/pen/GmqJEm

Thank you!

**Comment by desandro** [comment]
Sorry to hear you're having trouble with Masonry.

1. Looks like you're missing [`columnWidth`](http://masonry.desandro.com/options.html#columnwidth).
2. You're using Masonry JS, but you have a `masonry` object, like you're using Isotope. You can remove this object and move its options into the main object. 

See demo http://codepen.io/desandro/pen/8c8baf7734118150978e28cb50f1b9d1

**Comment by andru26** [comment]
Hello Desandro, thank you for the reply and demo.
But the problem is that the columns in my actual code do not have fixed width, so I can not use the columnWidth option.
Is it any way of achieving the correct layout with responsive columns?

Many thanks!

**Comment by desandro** [comment]
See http://masonry.desandro.com/layout.html#responsive-layouts

**Comment by andru26** [comment]
Hello,
I tried the suggested solution and I am was able to make it work, but I have one more issue.

Firstly, I would like to clarify my setup: My layout contains 300x250 ads in addition to the image elements
and currently I am not able to append the ads items to the masonry layout using the masonryImagesReveal function.
The ads look essentially like this:
<div class="grid-item ad-item">
    <div class="ad"></div>
  </div>

I created another pen, which shows the current code I using. In my case, the thumbnails are simply loaded from a hidded 'div.masonry-items' on the page: http://codepen.io/andru26/pen/EmNgWm

I really appreciate the help.
